pkgs/tools/filesystems/btrfs-dedupe/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchgit, rustPlatform, lzo, zlib }:
+
+with rustPlatform;
+
+buildRustPackage rec {
+  name = "btrfs-dedupe-${version}";
+  version = "1.0.3-git";
+
+  src = fetchgit {
```

Comment:
Does this rely on submodules or must cargo builds use git checkouts? If not, using source/release archives is preferred (better caching). `fetchFromGitHub` is a convenient method to do this.